Add a new section for runtime download components.

Arnab-Developer opened the pull request:
## Summary

It is clear from the introduction.md documentation file that what is included in SDK download but not clear what is included in runtime download. So I have made a new section and add some points for runtime download components in introduction.md file.

Fixes #22203

Arnab-Developer committed:
Add a new section for runtime download components.

It is clear from the introduction.md documentation file that what is included in SDK download but not clear what is included in runtime download. So I have made a new section and add some points for runtime download components in introduction.md file.

Arnab-Developer committed:
Fix markdown validation error.

tdykstra commented on docs/core/introduction.md:
```diff
@@ -139,6 +139,13 @@ The SDK download includes the following components:
 
+The runtime download includes the following components:
+
+* The desktop or ASP.NET Core runtime.
```
```suggestion
* Optionally, the desktop or ASP.NET Core runtime.
```

Arnab-Developer replied:
As a part of .NET runtime download, we need to download either ASP.NET Core runtime or desktop runtime. The first point is mandatory. So why we should make the first point as optional?

https://dotnet.microsoft.com/download/dotnet/current/runtime

tdykstra replied:
See this page:  https://dotnet.microsoft.com/download/dotnet/5.0

Arnab-Developer replied:
Thanks for the information. I have made the suggested changes.

tdykstra reviewed (approved):
@Arnab-Developer Thanks for submitting this PR! I made one suggestion.

Arnab-Developer committed:
Desktop or ASP.NET Core runtime download is optional.

Desktop or ASP.NET Core runtime download is optional. User can download only the runtime to run just a console app.

Made the changes based on PR review.

Co-authored-by: Tom Dykstra <tdykstra@microsoft.com>
